lib/react_client/react_interop.dart

```diff
@@ -34,8 +36,83 @@ abstract class React {
 
   external static bool isValidElement(dynamic object);
   external static ReactClass get Fragment;
+
+  external static JsRef createRef();
+}
+
+/// Creates a [Ref] object that can be attached to an element via the ref attribute.
+///
+/// __Example__:
+///
+///     class FooComponent extends react.Component2 {
+///       final Ref<BarComponent> barRef = createRef();
+///       final Ref<InputElement> inputRef = createRef();
+///
+///       render() => react.div({}, [
+///         Bar({'ref': barRef}),
+///         react.input({'ref': inputRef}),
+///       ]);
+///     }
+///
+/// Learn more: <https://reactjs.org/docs/refs-and-the-dom.html#creating-refs>.
+Ref<CurrentType> createRef<CurrentType>() {
+  return new Ref<CurrentType>();
+}
+
+/// The return type of [createRef], the constructor wraps [React.createRef] for use in
+/// Dart.
+///
+/// A [Ref] can be attached to an element via the ref attribute. When the element renders,
```

Comment:
Same as above regarding verbiage here.

Also, some Dart style guide tips:
- you don't need to reference `[Ref]` since that's the class your documenting: https://dart.dev/guides/language/effective-dart/documentation#avoid-redundancy-with-the-surrounding-context
- prefer using the word "this" within doc comments to refer to the current class instance https://dart.dev/guides/language/effective-dart/documentation#prefer-using-this-instead-of-the-to-refer-to-a-members-instance
- prefer brevity https://dart.dev/guides/language/effective-dart/documentation#prefer-brevity


Another idea for how to word this:
```dart
/// When this is provided as the ref prop, a reference to the rendered component 
/// will be available via [current].
```

Reply:
Thank you for the tips and references!